File: backend/src/main/java/org/luxons/sevenwonders/game/data/definitions/WonderSideDefinition.java
package org.luxons.sevenwonders.game.data.definitions;

import java.util.List;
import java.util.stream.Collectors;

import org.luxons.sevenwonders.game.Settings;
import org.luxons.sevenwonders.game.resources.ResourceType;
import org.luxons.sevenwonders.game.wonders.WonderStage;

// the fields are injected by Gson
@SuppressWarnings("unused,MismatchedQueryAndUpdateOfCollection")
class WonderSideDefinition {

    private ResourceType initialResource;

    private List<WonderStageDefinition> stages;

    private String image;

    ResourceType getInitialResource() {
        return initialResource;
    }

    List<WonderStage> createStages(Settings settings) {
        return stages.stream().map(def -> def.create(settings)).collect(Collectors.toList());
    }

    String getImage() {
        return image;
    }
}

File: game-engine/src/main/kotlin/org/luxons/sevenwonders/game/data/definitions/WonderSideDefinition.kt
package org.luxons.sevenwonders.game.data.definitions

import org.luxons.sevenwonders.game.resources.ResourceType
import org.luxons.sevenwonders.game.wonders.WonderStage

internal class WonderSideDefinition(
    val initialResource: ResourceType,
    private val stages: List<WonderStageDefinition>,
    val image: String
) {
    fun createStages(): List<WonderStage> {
        return stages.map { def -> def.create() }
    }
}
